gdrdrv/gdrdrv.c

```diff
@@ -479,16 +573,21 @@ static int gdrdrv_pin_buffer(gdr_info_t *info, void __user *_params)
         }
     }
 
-
     // here a typical driver would use the page_table to fill in some HW
     // DMA data structure
 
-    params.handle = mr->handle;
-
     mutex_lock(&info->lock);
-    list_add(&mr->node, &info->mr_list);
+    if (gdr_generate_mr_handle(info, mr) != 0) {
+        gdr_err("No address space left for BAR1 mapping.\n");
+        ret = -ENOMEM;
+    }
+
+    if (!ret)
+        list_add(&mr->node, &info->mr_list);
     mutex_unlock(&info->lock);
 
+    params.handle = mr->handle;
```

Comment:
purely for style, I'd rather access mr->handle when it has a sane value only, i.e. when there are no errors so after list_add().